arangod/IResearch/IResearchLink.cpp

```diff
@@ -239,48 +245,56 @@ std::ostream& operator<<(std::ostream& out, ThreadGroupStats const& stats) {
 /// @struct Task
 /// @brief base class for asynchronous maintenance tasks
 ////////////////////////////////////////////////////////////////////////////////
-template<typename T>
+template <typename T>
 struct Task {
   void schedule(std::chrono::milliseconds delay) const {
     LOG_TOPIC("eb0da", TRACE, arangodb::iresearch::TOPIC)
-        << "scheduled a " << T::typeName()
-        << " task for arangosearch link '" << id
-        << "', delay '" << delay.count() << "'";
+        << "scheduled a " << T::typeName() << " task for arangosearch link '"
+        << id << "', delay '" << delay.count() << "'";
 
     LOG_TOPIC("eb0d2", TRACE, arangodb::iresearch::TOPIC)
-        << T::typeName() << " pool: "
-        << ThreadGroupStats(async->stats(T::threadGroup()));
+        << T::typeName()
+        << " pool: " << ThreadGroupStats(async->stats(T::threadGroup()));
 
     if (!link->terminationRequested()) {
       async->queue(T::threadGroup(), delay, static_cast<const T&>(*this));
     }
   }
 
   std::shared_ptr<MaintenanceState> state;
-  IResearchFeature* async;
+  IResearchFeature* async{nullptr};
   IResearchLink::AsyncLinkPtr link;
   IndexId id;
-}; // Task
+};
+
+template <typename T>
+T getMetric(const IResearchLink& link) {
+  T metric;
+  metric.addLabel("viewId", std::move(link.getViewId()));
+  metric.addLabel("collId", std::move(link.getCollectionName()));
+  metric.addLabel("shardName", std::move(link.getShardName()));
+  metric.addLabel("dbName", std::move(link.getDbName()));
```

Comment:
it seems that `std::move(...)` isn't necessary given all getters return `std::string` by value

Reply:
Done.